File: Daily/PY/Leetcode938-二叉搜索树的范围和.py
class Solution:
    def __init__(self):
        self.res = 0
    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
        def dfs(root):
            if not root:
                return
            dfs(root.left)
            if root.val <= high and root.val >= low:
                self.res += root.val
            dfs(root.right)
        dfs(root)
        return self.res

File: Leetcode/sbgo/Leetcode938-二叉搜索树的范围和.go
package sbgo

func rangeSumBST(root *TreeNode, low int, high int) int {
	res := 0
	var dfs func(*TreeNode)
	dfs = func(root *TreeNode) {
		if root == nil {
			return
		}
		dfs(root.Left)
		if root.Val <= high && root.Val >= low {
			res += root.Val
		}
		dfs(root.Right)
	}
	dfs(root)
	return res
}
